### PR TITLE
(BOLT-997) Clean bash implementation for RedHat

### PR DESCRIPTION
**Pull request by donoghuc**
When `lsb_release` is used to find platform name in the bash implementation for some red-hat versions (example redhat-5-x86_64) the OS name fact reports `RedHatEnterpriseServer`. This commit changes shortens the output to `RedHat` as the puppet_agent::install task expects.

When a target is missing both `lsb_release` and `os-release` pick up the name and version from `redhat-release`.

### CODE REVIEW

**Comment by donoghuc** [comment]
Need to fix up acceptance tests.

**Comment by donoghuc** [comment]
Now passing acceptance tests in Jenkins CI. https://jenkins-master-prod-1.delivery.puppetlabs.net/view/All/job/forge-module_puppetlabs-facts_intn-sys_full-pe-current-adhoc/

File: spec/acceptance/init_spec.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 require 'beaker-task_helper/inventory'
 require 'bolt_spec/run'
 
-describe 'facts task' do
+describe 'facts task', unless: fact_on(default, 'os.release.full') == '2008 R2' do
```
[the remark this answers is not in the document]
Comment (donoghuc) [reply spec/acceptance/init_spec.rb]: Working on debugging this. Here is the error message:
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt/modules$ bolt task run facts -n example
Started on jz7jmgox4461ibq.delivery.puppetlabs.net...
Failed on jz7jmgox4461ibq.delivery.puppetlabs.net:
  The task failed with exit code 1:
  The term 'ruby.exe' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
  
  At line:2 char:2
  + & <<<<  ruby.exe '-S' '"C:\Users\Administrator\AppData\Local\Temp\dphe0jah.laj\ruby.rb"'
      + CategoryInfo          : ObjectNotFound: (ruby.exe:String) [], CommandNotFoundException
      + FullyQualifiedErrorId : CommandNotFoundException
 
  {
  }
Failed on 1 node: jz7jmgox4461ibq.delivery.puppetlabs.net
Ran on 1 node in 3.34 seconds
```
With the following inventory:
```
---
nodes:
  - name: jz7jmgox4461ibq.delivery.puppetlabs.net
    alias: example
    config:
      transport: winrm
      winrm:
        user: Administrator
        password: ****
        ssl: false
    features:
      - puppet-agent
```

[the remark this answers is not in the document]
Comment (MikaelSmith) [reply spec/acceptance/init_spec.rb]: Hmm. We should be setting up PATH so it looks for ruby.exe. I wonder if it gets installed somewhere unexpected.

**Pull request merged**